src/Cli.Tests/UpdateRuntimeTests.cs

```diff
@@ -0,0 +1,132 @@
+        public void TestNoUpdateRuntimeSettings()
+        {
+            _fileSystem!.AddFile(_testConfigFile, new MockFileData(INITIAL_CONFIG));
+
+            UpdateRuntimeOptions options = new(
+                depthLimit: null,
+                config: _testConfigFile
+            );
+
+            Assert.IsTrue(_fileSystem!.File.Exists(_testConfigFile));
+            Assert.IsTrue(TryUpdateRuntimeSettings(options, _runtimeConfigLoader!, _fileSystem!));
+
+            string updatedConfig = _fileSystem!.File.ReadAllText(_testConfigFile);
+            // Assert that INITIAL_CONFIG is same as the updated config
+            Assert.IsTrue(JToken.DeepEquals(JObject.Parse(INITIAL_CONFIG), JObject.Parse(updatedConfig)));
+        }
+
+        /// <summary>
+        /// Test to validate when no limit was specified, it gets added to the config file
+        /// using update runtime command. Here we are adding depth limit of 8 for GraphQL.
+        /// Once added it should be present in the config file.
+        [TestMethod]
+        public void TestAddDepthLimitForGraphQL()
+        {
+            _fileSystem!.AddFile(_testConfigFile, new MockFileData(INITIAL_CONFIG));
+            int maxDepthLimit = 8;
+
+            Assert.IsTrue(_fileSystem!.File.Exists(_testConfigFile));
+
+            // Initial State
+            Assert.IsTrue(RuntimeConfigLoader.TryParseConfig(INITIAL_CONFIG, out RuntimeConfig? config));
+            Assert.IsNotNull(config);
+            Assert.IsNull(config!.Runtime!.GraphQL!.DepthLimit);
+
+            // Add Depth Limit
+            UpdateRuntimeOptions options = new(
+                depthLimit: maxDepthLimit,
+                config: _testConfigFile
+            );
+            Assert.IsTrue(TryUpdateRuntimeSettings(options, _runtimeConfigLoader!, _fileSystem!));
+
+            // Assert after adding Depth Limit
+            string updatedConfig = _fileSystem!.File.ReadAllText(_testConfigFile);
+            Assert.IsTrue(RuntimeConfigLoader.TryParseConfig(updatedConfig, out config));
+            Assert.IsNotNull(config);
+            Assert.IsNotNull(config!.Runtime!.GraphQL!.DepthLimit);
+            Assert.AreEqual(maxDepthLimit, config!.Runtime!.GraphQL!.DepthLimit);
+        }
+
+        /// <summary>
+        /// Test to update the current depth limit for GraphQL and removal the depth limit using -1.
+        /// </summary>
+        [DataTestMethod]
+        [DataRow(20, DisplayName = "Update current depth limit for GraphQL.")]
+        [DataRow(-1, DisplayName = "Remove depth limit from GraphQL.")]
+        public void TestUpdateDepthLimitForGraphQL(int? newDepthLimit)
+        {
+            int currentDepthLimit = 8;
+
+            // Generate Config with Depth Limit
+            RuntimeConfigLoader.TryParseConfig(INITIAL_CONFIG, out RuntimeConfig? config);
+            Assert.IsNotNull(config);
+            config = config with
+            {
+                Runtime = config!.Runtime! with
```

Comment:
```suggestion
                Runtime = config.Runtime! with
```
config confirmed to be not null already. does intellisense still complain?